### ISSUE TITLE
Support for HyCC

### ISSUE THREAD

**Issue by benakhtarvt** [encryptogroup/MOTION#40]
I know you mentioned in your paper that you supported the HyCC with your MOTION code. It seems that it is still being cleaned up based off the documentation on the repository. I am currently a grad student working with MOTION for research and was wondering if it would be possible to have access to the HyCC adapter. It would be greatly appreciated.

**Comment by jonasschiller** [comment]
Hi, I would be hoping for the same thing.

**Comment by robinhundt** [comment]
Hi, maintenance for MOTION is unfortunately slow at the moment.
The MOTION2NX fork has code for a HyCC adapter here https://github.com/encryptogroup/MOTION2NX/blob/975a80d2d32c193969b596a9a05211c8881e0752/src/contrib/hycc/hycc_adapter.h .
However, as far as I know, MOTION2NX does not compile at the moment. I'm not sure whether it would be easier to fix MOTION2NX and get it to compile, or if it is straightforward to just integrate the HyCCAdapter of MOTION2NX into this main repository of MOTION.

**Comment by robinhundt** [comment]
Another option would be to use FUSE (https://github.com/encryptogroup/FUSE) as the intermediate format between HyCC and MOTION. @norakh can probably provide input on whether this conversion is currently working.

**Comment by norakh** [comment]
The HyCC Adapter from MOTION2NX is iirc a bit buggy and might need some more work to add into MOTION. I am working on an example in FUSE which I will add here once I am done that is hopefully easier to use and works.

**Comment by jonasschiller** [comment]
Thank you both for your replies. As i was looking at the examples in MOTION i saw that the AES one is already using a bristol format circuit similar to some operations such as the division. Can I just follow this structure with a Bristol Circuit from HyCC?

**Comment by norakh** [comment]
Sure, if that's already sufficient for your use case and you have the Bristol Format examples from HyCC at hand, this should do it.

**Comment by jonasschiller** [comment]
Thanks for all your help!
Do you know if there is a way to access the return of a Bristol Format algorithm similar to the AES example in the check function during the computation and not only after the results are opened?

**Comment by robinhundt** [comment]
Yes, you can use the `ShareWrappe::Evaluate` ([source](https://github.com/encryptogroup/MOTION/blob/master/src/motioncore/protocols/share_wrapper.h#L151)) function which returns a concatenated `ShareWrapper` of the outputs of Bristol circuit. These have not been opened, so they are indeed shares of the output. You can then use the ShareWrappers to perform additional operations on the output.

**Comment by jonasschiller** [comment]
Thanks for the answer but I'm afraid I didn't really pose my question right. 
For my program I would like to first perform a circuit with 300 times in parallel and then transform the results to one long std::vector<mo::BitVector<>> to put the result into another bristol circuit. 
If tried like this and some other ways but when I run it I seem to get an out of range error because it tries to access the values before the circuit is executed

auto result{input.Evaluate(logreg_algorithm)};
auto output = result.Unsimdify();
for (int t=0; t < 300;t++){
        for (int s=0;s<160;s++){
               output[i].As<mo::BitVector<>>().Get(s);
}       
}
Do know if there is a way to do this?
